### DISCUSSION TITLE
Add support for DDR for DoH

### DISCUSSION THREAD

**Discussion by cyberblob** [TechnitiumSoftware/DnsServer#705] [Ideas]
This allows devices to get the DoH servers to use for all devices on network.

"Here is how it works: Windows starts knowing only about a plain-text DNS server (knowing only an IP address for a resolver). Hoping to encrypt its traffic, Windows sends a DNS SVCB query for “resolver.arpa”, a new special use domain name that refers to the resolver receiving the query as opposed to a name within the DNS hierarchy. If an SVCB record is returned, it carries a DoH template along with either IP address hints or additional A/AAAA records for the DoH template name."

https://techcommunity.microsoft.com/t5/networking-blog/making-doh-discoverable-introducing-ddr/ba-p/2887289

https://datatracker.ietf.org/doc/draft-ietf-add-ddr/

**Comment by ShreyasZare** [comment]
Thanks for asking. The support for DDR already exists. You just need to add the SVCB record by creating a "resolver.arpa" primary zone that is mentioned in the article and it would work.

**Comment by Hemsby** [comment]
Please could you share an example on how to create the SVCB Record in this zone.

**Reply by ShreyasZare** [reply]
Sure. Just create `resolver.arpa` primary zone and add the SVCB record as shown below:

![image](https://github.com/TechnitiumSoftware/DnsServer/assets/10008172/2af626d8-63d2-49df-bc00-129409970078)

The above record will add SVCB record to indicate that the client should use `https://dns.quad9.net/dns-query` DoH service.
